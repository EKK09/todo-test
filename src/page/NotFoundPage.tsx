import React from 'react';
import styled from 'styled-components';
import { useHistory } from 'react-router-dom';
import { ROUTE_PATH } from 'src/route/routes';

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  color: DarkTurquoise;
`;
const Title = styled.div`
  font-size: 25vh;
`;
const Description = styled.h2`
  opacity: 0.4;
`;
const Button = styled.button`
  color: white;
  background-color: DarkTurquoise;
  font-size: larger;
  padding: 6px 12px;
  cursor: pointer;
  border-radius: 5px;
`;

const NotFoundPage = (): React.ReactElement => {
  const history = useHistory();

  const goToHomePage = () => {
    history.replace(ROUTE_PATH.HOME);
  };

  return (
    <Wrapper>
      <div>
        <Title>404</Title>
        <Description>Oops, Nothing here...</Description>
        <Button type="submit" onClick={goToHomePage}>Go Home</Button>
      </div>
    </Wrapper>
  );
};

export default NotFoundPage;
