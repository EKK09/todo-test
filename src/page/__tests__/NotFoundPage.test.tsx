import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { useHistory } from 'react-router-dom';
import NotFoundPage from '../NotFoundPage';

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn(),
}));
describe('NotFoundPage component', () => {
  it('Render NotFoundPage', async () => {
    const wrapper = render(<NotFoundPage />);

    expect(wrapper.container).toMatchSnapshot();
  });

  it('click Go home button', async () => {
    const mockReplace = jest.fn();
    (useHistory as jest.Mock).mockImplementation(() => ({ replace: mockReplace }));
    const { getByText } = render(<NotFoundPage />);
    const button = getByText('Go Home');
    fireEvent.click(button);

    expect(mockReplace).toBeCalledWith('/');
    expect(mockReplace).toBeCalledTimes(1);
  });
});
