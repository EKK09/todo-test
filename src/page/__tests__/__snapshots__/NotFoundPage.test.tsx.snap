// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component Render NotFoundPage 1`] = `
<div>
  <div
    class="sc-bdfBQB kBTvhN"
  >
    <div>
      <div
        class="sc-gsTEea dSrViT"
      >
        404
      </div>
      <h2
        class="sc-dlfnuX fHpSky"
      >
        Oops, Nothing here...
      </h2>
      <button
        class="sc-hKgJUU cQDtsj"
        type="submit"
      >
        Go Home
      </button>
    </div>
  </div>
</div>
`;
